ucp/_libs/ucp_py_buffer_helper.pyx

```diff
@@ -192,7 +192,8 @@ cdef class BufferRegion:
         self._populate_ptr(obj)
 
     cpdef _populate_ptr(self, format_[:] pyobj):
-        self.shape = pyobj.shape
+        # Notice, `len(memoryview.shape)` might not equal `memoryview.ndim`
+        self.shape = [pyobj.shape[i] for i in range(pyobj.ndim)]
```

Comment:
I believe this is expected to be a tuple elsewhere as opposed to a list

Reply:
I just noticed the tuple usage as well:
```python3

In [3]: np_arr = np.array([[1, 2], [3, 4]])                                                                            

In [4]: v = memoryview(np_arr)                                                                                         

In [5]: v.shape                                                                                                        
Out[5]: (2, 2)

In [6]: v2 = memoryview(b'abcdefg')                                                                                    

In [7]: v2.shape                                                                                                       
Out[7]: (7,)
```
In light of this, we should change the initialization of BufferRegion shape [here](https://github.com/rapidsai/ucx-py/blob/devel/ucp/_libs/ucp_py_buffer_helper.pyx#L114) from:
```
        self.shape = [0]
```
to
```
        self.shape = (0, )
```